src/nbl/asset/utils/CSPIRVIntrospector.cpp

```diff
@@ -150,32 +158,56 @@ NBL_API2 bool CSPIRVIntrospector::CPipelineIntrospectionData::merge(const CSPIRV
 //
 NBL_API2 core::smart_refctd_dynamic_array<SPushConstantRange> CSPIRVIntrospector::CPipelineIntrospectionData::createPushConstantRangesFromIntrospection(core::smart_refctd_ptr<const CStageIntrospectionData>& introspection)
 {
-    auto& ps = introspection->getPushConstants();
-    auto a = ps.type->memberInfoStorage;
+    auto& pc = introspection->getPushConstants();
 
-    core::vector<SPushConstantRange> tmp; tmp.reserve(MaxPushConstantsSize);
-    // Use Run Length Encode
-    /*SPushConstantRange prev = {
+    core::vector<SPushConstantRange> tmp; 
+    tmp.reserve(MaxPushConstantsSize);
+    
+    SPushConstantRange prev = {
         .stageFlags = m_pushConstantBytes[0].value,
         .offset = 0,
         .size = 0
     };
-    for ()
+
+    SPushConstantRange current = {
+        .offset = 0,
+        .size = 0
+    };
```

Comment:
I'd maybe keeptrack only of currentStage and currentOffset, the size (which is redundant) being kept around makes for difficult reading